src/public/providers.ts

```diff
@@ -4,21 +4,52 @@ const providerList: HTMLElement = helper.getElement('.providerList')
 type Provider = {
   id?: string
   name: string
-  api: string
+  service: string
   keys: string[]
 }
 
-/* HardCoded dns object to display mock UI */
-const temp = {
-  dnsGd: {
-    name: 'GoDaddy',
-    api: 'https://api.godaddy.com',
-    keys: ['GD_Key', 'GD_Secret']
-  },
-  dnsPdns: {
-    name: 'PowerDNS',
-    api: 'https://api.powerdns.com',
-    keys: ['PDNS_Url', 'PDNS_ServerId', 'PDNS_Token', 'PDNS_Ttl']
+class ProviderKey {
+  providerKey: string
+  providerKeyValue: string
+  providerKeysContainer: HTMLElement
+  constructor(
+    providerKey: string,
+    providerKeyValue: string,
+    providerName: string,
+    providerKeysContainer: HTMLElement
+  ) {
+    this.providerKey = providerKey
+    this.providerKeysContainer = providerKeysContainer
+    this.providerKeyValue = providerKeyValue
+    const providerKeyElement = document.createElement('div')
+    providerKeyElement.innerHTML = `
+      <span class="providerKeyName">${this.providerKey}</span>
+      <input type="text" value="${this.providerKeyValue}" class="keyInput"></input>
+      <button type="button" class="btn btn-primary saveOrEditKeysButton">Save</button>      
+    `
+    const saveOrEdit = helper.getElement(
+      '.saveOrEditKeysButton',
+      providerKeyElement
+    )
+    saveOrEdit.onclick = (): void => {
+      const keyName = helper.getElement('.providerKeyName', providerKeyElement)
```

Comment:
you don't need to get `keyName` element, because you already have it! `this.providerKey`